src/new-client/src/App/Trades/TradesGrid/TableHeadCell.tsx

```diff
@@ -34,42 +43,104 @@ const TableHeadCell = styled.th`
     display: inline-block;
   }
 `
+const AlignedFilterIcon = styled(FaFilter)`
+  margin-left: 0.2rem;
+  margin-right: 0.1rem;
+`
+
+export const ContactUsPopup = styled(Popup)`
+  box-shadow: 0 7px 26px 0 rgba(23, 24, 25, 0.5);
+  bottom: calc(2rem + 0.25rem);
+  border-radius: 0.5rem;
+`
 
 export const TableHeadCellContainer: React.FC<{
   field: ColField
 }> = ({ field }) => {
   const [showFilter, setShowFilter] = useState(false)
-  const ref = useRef<HTMLDivElement>(null)
+  const ref = useRef<HTMLTableHeaderCellElement>(null)
   const { displayMenu, setDisplayMenu } = usePopUpMenu(ref)
   const tableSort = useTableSort()
   const { headerName, filterType } = colConfigs[field]
+  const numeric = filterType === "number" && field !== "tradeId"
+  const [comparator, setComparator] = useState("Equals")
```

Comment:
This is a bit of a smell.  We shouldn't have to care about the other applied filters (set/date) at this level.  We should just instantiate the right component (NumberFilter//SetFilter/DateFilter) based on the configured filter type for that field and let the component invoke the hook and get the options.  I realize you are solving the bug we had.  But I think we need another solution.  @josepot ?